src/ast.rs

```diff
@@ -301,6 +301,17 @@ impl VarScope {
         Self { names: Vec::new() }
     }
 
+    /// Return the variable with the given name, if one exists.
+    pub fn get(&self, name: &str) -> Option<Var> {
```

Comment:
Could you also use this `get` function in `get_or_insert` then? Since that part is pretty much identical, there's just the additional `unwrap_or_else`..

Reply:
Oh, I didn't even notice. It's funny that a part of the code was written in very different styles. I'm not sure which I like better.

Fixed.